bee-message/src/milestone/index.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_common::packable::{Packable, Read, Write};
+
+use core::ops::{Add, Deref, Sub};
+
+/// A wrapper around a `u32` that represents a milestone index.
+#[derive(Debug, Clone, Copy, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct MilestoneIndex(pub u32);
```

Comment:
I suggest adding `#[repr(transparent)]` to this type. It really is just a `u32`.

Reply:
Too bad the `Step` trait is nightly, because I wanted to iterate over `MilestoneIndex`es so often during the pruning impl.